src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,34 +35,42 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
+        Matcher matcher = ADDRESS_VALIDATION_REGEX.matcher(address);
+        if (!matcher.matches()) {
```

Comment:
It is brilliant to use Matcher, but remember that this is still a specific way to do validation. So you still need to call the encapsulated method `isValidAddress(...)` here.